ga4gh/backend.py

```diff
@@ -209,6 +214,83 @@ def getReadGroupSets(self):
         """
         return list(self._readGroupSetIdMap.values())
 
+    def getReferenceSets(self):
+        """
+        Returns the list of ReferenceSets in this backend
+        """
+        return list(self._referenceSetIdMap.values())
+
+    def getReference(self, id_):
+        """
+        Returns a reference with the given id_
+        """
+        return self.runGetRequest(self._referenceIdMap, id_)
+
+    def getReferenceSet(self, id_):
+        """
+        Returns a referenceSet with the given id_
+        """
+        return self.runGetRequest(self._referenceSetIdMap, id_)
+
+    def listReferenceBases(self, id_, requestArgs):
+        # parse arguments
+        try:
+            reference = self._referenceIdMap[id_]
+        except KeyError:
+            raise exceptions.ObjectWithIdNotFoundException(id_)
+        start = 0
+        end = datamodel.PysamDatamodelMixin.fastaMax
```

Comment:
Is there anyway that we can use the validation machinery here to avoid some of this messiness? Can't we create a `ListReferenceBasesRequest` object out of the requestArgs dictionary? We then have full validation, we reuse the general machinery for parsing and life is good.